txs/tx_utils.go

```diff
@@ -37,65 +33,73 @@ func NewSendTx() *SendTx {
 	}
 }
 
-func (tx *SendTx) AddInput(st AccountGetter, pubkey crypto.PubKey, amt int64) error {
+func (tx *SendTx) AddInput(st acm.Getter, pubkey acm.PublicKey, amt uint64) error {
 	addr := pubkey.Address()
-	acc := st.GetAccount(addr)
+	acc, err := st.GetAccount(addr)
+	if err != nil {
+		return err
+	}
 	if acc == nil {
-		return fmt.Errorf("Invalid address %X from pubkey %X", addr, pubkey)
+		return fmt.Errorf("invalid address %s from pubkey %s", addr, pubkey)
 	}
-	return tx.AddInputWithNonce(pubkey, amt, acc.Sequence+1)
+	return tx.AddInputWithNonce(pubkey, amt, acc.Sequence()+1)
 }
 
-func (tx *SendTx) AddInputWithNonce(pubkey crypto.PubKey, amt int64, nonce int) error {
+func (tx *SendTx) AddInputWithNonce(pubkey acm.PublicKey, amt uint64, sequence uint64) error {
 	addr := pubkey.Address()
 	tx.Inputs = append(tx.Inputs, &TxInput{
 		Address:   addr,
 		Amount:    amt,
-		Sequence:  nonce,
-		Signature: crypto.SignatureEd25519{},
+		Sequence:  sequence,
+		Signature: crypto.SignatureEd25519{}.Wrap(),
 		PubKey:    pubkey,
 	})
 	return nil
 }
 
-func (tx *SendTx) AddOutput(addr []byte, amt int64) error {
+func (tx *SendTx) AddOutput(addr acm.Address, amt uint64) error {
 	tx.Outputs = append(tx.Outputs, &TxOutput{
 		Address: addr,
 		Amount:  amt,
 	})
 	return nil
 }
 
-func (tx *SendTx) SignInput(chainID string, i int, privAccount *acm.PrivAccount) error {
+func (tx *SendTx) SignInput(chainID string, i int, privAccount acm.PrivateAccount) error {
 	if i >= len(tx.Inputs) {
 		return fmt.Errorf("Index %v is greater than number of inputs (%v)", i, len(tx.Inputs))
 	}
-	tx.Inputs[i].PubKey = privAccount.PubKey
-	tx.Inputs[i].Signature = privAccount.Sign(chainID, tx)
+	tx.Inputs[i].PubKey = privAccount.PublicKey()
+	tx.Inputs[i].Signature = acm.ChainSign(privAccount, chainID, tx)
 	return nil
 }
 
 //----------------------------------------------------------------------------
 // CallTx interface for creating tx
 
-func NewCallTx(st AccountGetter, from crypto.PubKey, to, data []byte, amt, gasLimit, fee int64) (*CallTx, error) {
+func NewCallTx(st acm.Getter, from acm.PublicKey, to *acm.Address, data []byte,
+	amt, gasLimit, fee uint64) (*CallTx, error) {
+
 	addr := from.Address()
-	acc := st.GetAccount(addr)
+	acc, err := st.GetAccount(addr)
+	if err != nil {
+		return nil, err
+	}
 	if acc == nil {
-		return nil, fmt.Errorf("Invalid address %X from pubkey %X", addr, from)
+		return nil, fmt.Errorf("invalid address %s from pubkey %s", addr, from)
 	}
 
-	nonce := acc.Sequence + 1
+	nonce := acc.Sequence() + 1
 	return NewCallTxWithNonce(from, to, data, amt, gasLimit, fee, nonce), nil
 }
 
-func NewCallTxWithNonce(from crypto.PubKey, to, data []byte, amt, gasLimit, fee int64, nonce int) *CallTx {
-	addr := from.Address()
+func NewCallTxWithNonce(from acm.PublicKey, to *acm.Address, data []byte,
```

Comment:
`NewCallTxWithSequence` for consistency?

Reply:
Casey you need to be stopped. Consistency this. Consistency that. It's globalisation gone mad. Go on then... There's a few other usages of nonce I'll change too.